lms/djangoapps/courseware/module_render.py

```diff
@@ -490,12 +478,14 @@ def make_xqueue_callback(dispatch='score_update'):
     # TODO: Queuename should be derived from 'course_settings.json' of each course
     xqueue_default_queuename = descriptor.location.org + '-' + descriptor.location.course
 
-    xqueue = {
-        'interface': XQUEUE_INTERFACE,
-        'construct_callback': make_xqueue_callback,
-        'default_queuename': xqueue_default_queuename.replace(' ', '_'),
-        'waittime': settings.XQUEUE_WAITTIME_BETWEEN_REQUESTS
-    }
+    xqueue_service = XQueueService(
+        construct_callback=make_xqueue_callback,
```

Comment:
There's a lot of rigamarole that goes into [creating the default `xqueue_callback_url_prefix`](https://github.com/edx/edx-platform/pull/29312/files#diff-3b0c1e091e61826c4448bc5924e2911dedfd829754b2c6087e7067dfbdeed895R367) from the `request`, and then to pass it around through all the methods so that it's available when we want to call `make_xqueue_callback`. 

Instead of all of this, can we just use `settings,LMS_ROOT_URL` as the default (still allowing the user to override with `settings.XQUEUE_INTERFACE["callback_url"]`)?

Reply:
Hmm, will that work for testing in preview as well? 🤔

Reply:
Hmm.. I think so. The [`xqueue_callback`](https://github.com/edx/edx-platform/blob/ad5ad72273ce36bb3b14b5b939ea5bacac78d88f/lms/djangoapps/courseware/module_render.py#L955) function doesn't require authentication -- it receives everything it needs from the URL -- user_id, course_id, etc -- and so it shouldn't matter that the request is being received from the preview service to the LMS.

Reply:
My understanding is that https://preview.pr29312.sandbox.opencraft.hosting/ is served by the lms process as well and since the solution is posted to the preview subdomain when the author is on preview, because `get_xqueue_callback_url_prefix()` is looking at `request.get()`, the callback URL will be to https://preview.pr29312.sandbox.opencraft.hosting. 

If instead, the code always uses `settings.LMS_ROOT_URL`, the callback url will always point to the main LMS domain, right? If these assumptions are correct, will everything continue to work as expected?

Reply:
* Yes, https://preview.pr29312.sandbox.opencraft.hosting/ is served by the lms process
* Yes, if we use `settings.LMS_ROOT_URL` as the default prefix, then we'll always be pointing to the main LMS domain for the xqueue callback, even when we're viewing the problem on preview.
* And from the code, the xqueue callback will work the same whether it's called from LMS or Preview.

I did some testing with the sandbox, and it looks good -- so am going to push a commit to make this change, so it can be fully tested.

Reply:
See 65db493